### PR TITLE
Add ln service error switch

### PR DESCRIPTION
**Pull request by vindard**
## Description

This PR is to:
- fix a small leak where domain errors were getting through to the app layer for unhandled errors in our error map (see **Screenshot**)
- to introduce a pattern for parsing and handling known errors from the lnd service library

---

### Screenshot
image

### CODE REVIEW

File: src/graphql/error-map.ts [hunk 1]
```diff
@@ -106,6 +107,10 @@ export const mapError = (error: ApplicationError): CustomApolloError => {
         "Too many onchain addresses creation, please wait for a while and try again."
       return new TooManyRequestError({ message, logger: baseLogger })
 
+    case "CouldNotFindPhoneCodeError":
+      message = "Invalid or incorrect phone code entered."
```
Comment (nicolasburtey) [inline src/graphql/error-map.ts:111]: do we really need to override the message? maybe we could just set a default message of `PhoneCodeError` as this one?

**Review by nicolasburtey — APPROVED** [review]

File: src/services/lnd/index.ts [hunk 2]
```diff
@@ -418,13 +454,20 @@ const lookupPaymentByPubkeyAndHash = async ({
...
-    return new UnknownLightningServiceError(err)
+    const errDetails = parseLndErrorDetails(err)
+    switch (errDetails) {
+      default:
+        return new UnknownLightningServiceError(err)
+    }
   }
 }
 
 const parseLndErrorDetails = (err) =>
```
Comment (nicolasburtey) [inline src/services/lnd/index.ts:465]: some documentation could be helpful here (maybe on the way `ln-service`/`lightning` manages error in a non conventional way.

**Review by dolcalmi — APPROVED** [review]
looks good, although this 'pattern' could be a generic parse function
```
    const errDetails = parseLndErrorDetails(err)
    switch (errDetails) {
      default:
        return new UnknownLightningServiceError(err)
    }
```

**Comment by vindard** [comment]
> looks good, although this 'pattern' could be a generic parse function

I was thinking that too, but was also thinking there could be cases where the same lnd error could lead to different domain errors e.g. `""insufficient local balance"` from lnd resolving to either `InsufficientBalanceForRoutingError` or `InsufficientBalanceForPaymentError` depending on the service method it is thrown from

File: src/services/lnd/index.ts (hunk 2 above)
Comment (vindard) [reply src/services/lnd/index.ts:465]: Added: https://github.com/GaloyMoney/galoy/pull/804

**Pull request merged**